xgcm/grid.py

```diff
@@ -178,13 +178,22 @@ def __init__(
             # TODO (Julius in #568) full hierarchy of conventions here
             # but override with any user-given options
 
+            # try sgrid parsing
+            if sgrid.assert_valid_sgrid(ds):
```

Comment:
I really like the way of checking the validity. Very readable!

Reply:
Thanks, see discussion in https://github.com/xgcm/xgcm/issues/566#issuecomment-1410690075 for related query about doing this for CF-data.